src/lua/utils.c

```diff
@@ -735,6 +735,30 @@ luaT_newthread(struct lua_State *L)
 	return L1;
 }
 
+/**
+ * A safer os.getenv() variant. The difference is that it returns the value only
+ * if it has sane length (see src/lib/core/utils.c for details).
+ */
+static int
+luaT_getenv(struct lua_State *L)
+{
+	if (lua_gettop(L) != 1 || !lua_isstring(L, 1))
+		return luaL_error(L, "usage: os.getenv(name)");
+	const char *name = lua_tostring(L, 1);
+	char *envvar = getenv_safe(name, NULL, 0);
+	if (envvar != NULL) {
+		lua_pushstring(L, envvar);
```

Comment:
If Lua OOM will hit us here, we'll loss the malloc'ed memory. Don't know, whether this theoretical problem worth any API/implementation complication, just noted.

Reply:
Unfortunately there are lots of places like this in our code, where we can leak memory.

<details><summary>To name a few</summary

https://github.com/tarantool/tarantool/blob/256da01046da8f4fbe81d629cf54ae32dc04a1fb/src/lua/fio.c#L507
https://github.com/tarantool/tarantool/blob/256da01046da8f4fbe81d629cf54ae32dc04a1fb/src/lua/httpc.c#L116
https://github.com/tarantool/tarantool/blob/256da01046da8f4fbe81d629cf54ae32dc04a1fb/src/lua/socket.c#L994

</details>

I think we should make Lua panic on OOM, because the user application will probably break in this case. With GC64 OOM shouldn't occur, anyway.

Reply:
Okay, let's consider it as negligible.